crates/sparrow-main/tests/e2e/collection_tests.rs

```diff
@@ -1,119 +1,49 @@
-                .map(|(index, data_type)| Field::new(format!("e{index}"), data_type, true)),
-        );
-
-        Arc::new(Schema::new(fields))
-    };
-
-    // Create the reader
-    let reader = std::io::Cursor::new(json.as_bytes());
-    let reader = arrow::json::ReaderBuilder::new(schema).build(reader)?;
-
-    // Read all the batches and concatenate them.
-    let batches: Vec<_> = reader.try_collect()?;
-    let read_schema = batches.get(0).context("no batches read")?.schema();
-    let batch = arrow::compute::concat_batches(&read_schema, &batches)
-        .context("concatenate read batches")?;
-
-    Ok(batch)
-}
-
-async fn json_to_parquet_file(json_input: &str, file: File) {
-    let fields = Fields::from(vec![
-        Field::new("key", DataType::Utf8, false),
-        Field::new("value", DataType::Int64, false),
-    ]);
-
-    let m1 = Arc::new(Field::new("map", DataType::Struct(fields), false));
-    let column_types = vec![
-        DataType::Map(m1, false),
-        DataType::Int64,
-        DataType::Float64,
-        DataType::Utf8,
-    ];
-    let record_batch = batch_from_json(json_input, column_types).unwrap();
-
-    // Create a Parquet writer
-    let mut writer = ArrowWriter::try_new(file, record_batch.schema(), None).unwrap();
-    writer.write(&record_batch).unwrap();
-
-    // Close the writer to finish writing the file
-    writer.close().unwrap();
+#[tokio::test]
+async fn test_get_dynamic_key() {
+    insta::assert_snapshot!(QueryFixture::new("{ value: Input.e0 | get(Input.e3) }").run_to_csv(&collection_data_fixture().await).await.unwrap(), @r###"
+    _time,_subsort,_key_hash,_key,value
+    1970-01-01T00:00:00.000002000,5039430902799166705,2359047937476779835,1,0
+    1970-01-01T00:00:00.000003000,5039430902799166706,2359047937476779835,1,10
+    1970-01-01T00:00:00.000003000,5039430902799166707,2359047937476779835,1,
+    1970-01-01T00:00:00.000003000,5039430902799166708,2359047937476779835,1,13
+    1970-01-01T00:00:00.000004000,5039430902799166709,2359047937476779835,1,11
+    "###);
 }
 
-async fn arrow_collection_data_fixture() -> DataFixture {
-    let mut path = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
-    path.pop();
-    path.pop();
-    path.push("testdata");
-    path.push("parquet/data_with_map.parquet");
-
-    let file = File::create(path).unwrap();
-    json_to_parquet_file(JSON_INPUT, file).await;
-
-    DataFixture::new()
-        .with_table_from_files(
-            TableConfig::new_with_table_source("Input", &Uuid::new_v4(), "time", None, "key", ""),
-            &["parquet/data_with_map.parquet"],
-        )
-        .await
-        .unwrap()
+#[tokio::test]
+async fn test_supplant() {
+    insta::assert_snapshot!(QueryFixture::new("{ value: get(\"airTemperature\" as large_string, Input.readings) | when(is_valid($input)) } ").with_dump_dot("asdf").run_to_csv(&supplant_data_fixture().await).await.unwrap(), @r###"
+    _time,_subsort,_key_hash,_key,value
+    "###);
```

Comment:
TODO: Remove this

The test works, and produces results as expected. However, we likely don't support `LargeUtf8` in other evaluators, meaning we probably can't manipulate the `keys` in any way. Also, having to cast to `as large_string` is weird, but unclear why the literal string isn't being inferred as `large_string` at the moment.